### PR TITLE
Fetch issue comments

### PR DESCRIPTION
**Pull request by mvolpato**
- Adds `GET /repos/:owner/:repo/issues/comments` for #38 .
- Adds `Reactions` to `Comment`. To get `Reactions` we need to add a custom header:
- Adds custom headers for GitHub API previews.
- Adds `application/vnd.github.squirrel-girl-preview` [custom header (Reactions)](https://developer.github.com/changes/2016-05-12-reactions-api-preview/).
- Tests updated and/or added.

### CODE REVIEW

File: OctoKit/Issue.swift [hunk 1]
```diff
@@ -226,6 +227,29 @@ public extension Octokit {
...
+    /// - Parameters:
+    /// - session: RequestKitURLSession, defaults to URLSession.sharedSession()
+    /// - owner: The user or organization that owns the repository.
+    /// - repository: The name of the repository.
+    /// - number: The number of the issue.
+    /// - page: Current page for comments pagination. `1` by default.
+    /// - perPage: Number of comments per page. `100` by default.
+    /// - completion: Callback for the outcome of the fetch.
+    @discardableResult
+    func commentsIssue(_ session: RequestKitURLSession = URLSession.shared, owner: String, repository: String, number: Int, page: String = "1", perPage: String = "100", completion: @escaping (_ response: Response<[Comment]>) -> Void) -> URLSessionDataTaskProtocol? {
```
Comment (pietbrauer) [inline OctoKit/Issue.swift:241]: Could you rename this to `issueComments` as it's the comments of an issue not the issue of a list of comments, if that makes sense.
```suggestion
    func issueComments(_ session: RequestKitURLSession = URLSession.shared, owner: String, repository: String, number: Int, page: String = "1", perPage: String = "100", completion: @escaping (_ response: Response<[Comment]>) -> Void) -> URLSessionDataTaskProtocol? {
```

File: OctoKit/Issue.swift [hunk 2]
```diff
@@ -237,6 +261,7 @@ enum IssueRouter: JSONPostRouter {
     case postIssue(Configuration, String, String, String, String?, String?, [String])
     case patchIssue(Configuration, String, String, Int, String?, String?, String?, Openness?)
     case commentIssue(Configuration, String, String, Int, String)
+    case readCommentsIssue(Configuration, String, String, Int, String, String)
```
Comment (pietbrauer) [inline OctoKit/Issue.swift:264]: Same here, please rename accordingly.

File: OctoKit/PreviewHeader.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    public var header: HTTPHeader {
+        switch self {
+        case .reactions:
+            return HTTPHeader(headerField: "Accept", value: "application/vnd.github.squirrel-girl-preview")
+        }
+    }
+}
+
+
```
Comment (pietbrauer) [inline OctoKit/PreviewHeader.swift:19]: Please remove the empty lines to have only one at the end

File: OctoKit/PreviewHeader.swift [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import Foundation
+import RequestKit
+
+/// Some API provide additional data for new (preview) API if a custom header is added to the request.
+///
+/// - Note: Preview APIs are subject to change.
+public enum PreviewHeader {
```
Comment (pietbrauer) [inline OctoKit/PreviewHeader.swift:7]: Nice one! 👏

File: OctoKit/Reactions.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/// `Reactions` are available on all Issues and Pull Requests.
+///
+/// `Reactions` are in general available as an object in some API response,
+/// there is no API available directly for `Reactions`.
+public struct Reactions: Codable {
+    public let totalCount: Int?
+    public let plus1: Int?
```
Comment (pietbrauer) [inline OctoKit/Reactions.swift:7]: I'm on the fence here. This looks super weird but maps the response correctly. Maybe thumbs up/down is more readable?

**Review by pietbrauer — CHANGES_REQUESTED** [review]
Nice work. Added some opinions, open for discussion 😊

File: OctoKit/Reactions.swift (hunk 5 above)
Comment (mvolpato) [reply OctoKit/Reactions.swift:7]: Your suggestion makes sense. I do not think GitHub will change the syntax of `+1` to something different than the thumbs-up.

File: OctoKit/PreviewHeader.swift (hunk 4 above)
Comment (mvolpato) [reply OctoKit/PreviewHeader.swift:7]: Thanks!

**Comment by pietbrauer** [comment]
This one has merge conflicts now 😬 Could you resolve them?

**Review by pietbrauer — APPROVED** [review]

**Comment by pietbrauer** [comment]
Thanks again!

**Pull request merged**